src/sage/combinat/permutation.py

```diff
@@ -5383,6 +5383,41 @@ def shifted_shuffle(self, other):
         return self.shifted_concatenation(other, "right").\
         right_permutohedron_interval(self.shifted_concatenation(other, "left"))
 
+    def transposition_distance(self, other):
+        r"""
+        Compute the transposition distance between two permutations.
+
+        INPUT:
+
+        - ``other`` -- Another permutation to compute the transposition distance to.
+
+        OUTPUT:
+
+        - An integer representing the transposition distance between the two permutations.
+
+        EXAMPLES::
+
+            sage: p1 = Permutation([2, 1, 3, 4])
+            sage: p2 = Permutation([1, 3, 2, 4])
+            sage: p1.transposition_distance(p2)
+            2
+        """
+        other = self.parent()(other)
+
+        n = self.parent().degree()
+        if n != other.parent().degree():
+            raise ValueError("permutations must have the same degree")
+
+        transpositions = 0
+        perm1 = list(self)
+
+        for i in range(n):
+            if perm1[i] != other[i]:
+                j = perm1.index(other[i], i)
+                perm1[i], perm1[j] = perm1[j], perm1[i]
+                transpositions += 1
+
+        return transpositions
```

Comment:
```suggestion
        return ZZ(transpositions)
```
This should return a Sage `Integer`, not a Python `int`.